### PR TITLE
Add new `Style/NumericPredicate` cop

### PR DESCRIPTION
**Pull request by Drenmi**
This change adds a new cop that checks for comparisons like `numeric > 0`, `numeric != 0`, etc. and suggests replacing them with their respective predicate methods:

 - `numeric == 0` -> `numeric.zero?`
 - `numeric != 0` -> `numeric.nonzero?`

Only when `TargetRubyVersion` >= 2.3:

 - `numeric > 0` -> `numeric.positive?`
 - `numeric < 0` -> `numeric.negative?`

This is arguably a controversial cop, because you can compare equality of things like `nil` or a `string` to a numeric (and get a nonsensical result), but I think that makes an even stronger case for this cop, as doing that means you're either relying on happenstance behaviour, or you have a silent bug in your code.

(This is not a problem for `#positive?` and `#negative?`, as most non-numerics don't respond to `#>` and `#<`.)

The cop is configurable to use either style,

### CODE REVIEW

**Comment by mikegee** [comment]
The style guide may need some updates if this is adopted: https://github.com/bbatsov/ruby-style-guide#predicate-methods

**Comment by Drenmi** [comment]
Thanks for the link @mikegee! I somehow completely missed this. Maybe we should make the style configurable, then.

**Comment by bbatsov** [comment]
> Thanks for the link @mikegee! I somehow completely missed this. Maybe we should make the style configurable, then.

Yeah, I'd say this should be configurable. I see the benefit of having consistency in your comparisons, although I'd definitely would not say that one style is better than the other.

**Comment by Drenmi** [comment]
The benefits are that the predicate methods will catch accidental comparisons with other types, most importantly `nil`, and that they can be easily used with e.g. `#map`, thanks to `Symbol#to_proc`. Whether this is enough to make it "better" is subjective however, so I finally found the time to make it configurable. 😀 

Also rebased.

File: config/enabled.yml [hunk 1]
```diff
@@ -552,6 +552,10 @@ Style/NumericLiteralPrefix:
   StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#numeric-literal-prefixes'
   Enabled: true
 
+Style/NumericPredicate:
+  Description: 'Use numeric predicate methods instead of comparison operators.'
```
Comment (bbatsov) [inline config/enabled.yml]: This has to be updated now that the cop is configurable.

**Comment by Drenmi** [comment]
@bbatsov: Updated.

**Comment by andyw8** [comment]
It seems there was no entry in the CHANGELOG for this new cop?

**Comment by Drenmi** [comment]
@andyw8: It seems like you're right. That's quite an oversight.

@bbatsov: Please advise.

**Comment by bbatsov** [comment]
There's no harm to add it now.

**Comment by Drenmi** [comment]
You're absolutely right @andyw8. I normally tick them off and then remove it to highlight my PR message, but this time was subject to a severe brain fart. 😊

**Comment by Drenmi** [comment]
@andyw8: After some looking around, it seems I destroyed it when I rebased to current `master`.

**Pull request merged**